client/web/compose/src/components/PageBlocks/Navigation/NavTypes/Compose.vue

```diff
@@ -0,0 +1,141 @@
+<template>
+  <tr>
+    <td />
+    <td>
+      <b-form-group :label="$t('navigation.fieldLabel')">
+        <b-form-input
+          v-model="value.options.itemOption.label"
+          type="text"
+          size="sm"
+        />
+      </b-form-group>
+    </td>
+    <td>
+      <b-form-group :label="$t('navigation.composePage')">
+        <b-form-select
+          v-model="value.options.itemOption.referenceId"
```

Comment:
wouldn't this be the pageID?
Also it would probably make sense to replace the b-form-select with vue select.
You can look at namespace sidebar for a good example